src/allocator/allocator.c

```
@@ -263,14 +262,38 @@ int free_raw_async(CUdeviceptr dptr, CUstream hStream) {
     return tmp;
 }
 
+/* Track an allocated async chunk and account it against its pool's high-water
+ * limit. Call with mutex held. */
+static int account_async_chunk(allocated_list_entry *e, size_t size, CUmemoryPool pool) {
+    size_t poollimit;
+    CUresult res = CUDA_OVERRIDE_CALL(cuda_library_entry,cuMemPoolGetAttribute,pool,CU_MEMPOOL_ATTR_RESERVED_MEM_HIGH,&poollimit);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)